### PR TITLE
Updating docs to use CollectionConverters over JavaConverters

### LINKED ISSUE #1644: 2.13 collection conversion documentation mentions deprecated import
In https://docs.scala-lang.org/overviews/collections-2.13/conversions-between-java-and-scala-collections.html we have the block indicating the converters import as `import collection.JavaConverters._` which triggers a deprecation warning.
As the documentation is 2.13 focused, it would be better to indicate `import scala.jdk.CollectionConverters._` instead.

### PR DESCRIPTION
**Pull request by zachhweed**
This fixes #1644

### CODE REVIEW

File: _overviews/collections-2.13/conversions-between-java-and-scala-collections.md [hunk 1]
```diff
@@ -27,8 +27,8 @@ Sometimes you might need to pass from one collection framework to the other. For
 
 To enable these conversions, simply import them from the [JavaConverters](https://www.scala-lang.org/api/{{ site.scala-version }}/scala/collection/JavaConverters$.html) object:
```
Comment (Philippus) [inline _overviews/collections-2.13/conversions-between-java-and-scala-collections.md]: To make this PR complete this line and the second paragraph of the page should also be changed.

File: _ja/overviews/collections/conversions-between-java-and-scala-collections.md [hunk 2]
```diff
@@ -25,8 +25,8 @@ Scala と同様に、Java
 
 このような変換を作動させるには、[JavaConverters](https://www.scala-lang.org/api/{{ site.scala-version }}/scala/collection/JavaConverters$.html) オブジェクトからインポートするだけでいい:
 
-    scala> import collection.JavaConverters._
-    import collection.JavaConverters._
+    scala> import scala.jdk.CollectionConverters._
+    import scala.jdk.CollectionConverters._
```
Comment (Philippus) [inline _ja/overviews/collections/conversions-between-java-and-scala-collections.md]: Only the overview for the scala 2.13 collections should be changed I think, the overview for the old collections should stay as is.

File: _overviews/collections/conversions-between-java-and-scala-collections.md [hunk 3]
```diff
@@ -27,8 +27,8 @@ Sometimes you might need to pass from one collection framework to the other. For
 
 To enable these conversions, simply import them from the [JavaConverters](https://www.scala-lang.org/api/{{ site.scala-212-version }}/scala/collection/JavaConverters$.html) object:
 
-    scala> import collection.JavaConverters._
-    import collection.JavaConverters._
+    scala> import scala.jdk.CollectionConverters._
+    import scala.jdk.CollectionConverters._
```
Comment (Philippus) [inline _overviews/collections/conversions-between-java-and-scala-collections.md]: Only the overview for the scala 2.13 collections should be changed I think, the overview for the old collections should stay as is.

File: _zh-cn/overviews/collections/conversions-between-java-and-scala-collections.md [hunk 4]
```diff
@@ -25,8 +25,8 @@ language: zh-cn
 
 使用这些转换很简单，只需从JavaConverters对象中import它们即可。
 
-    scala> import collection.JavaConverters._
-    import collection.JavaConverters._
+    scala> import scala.jdk.CollectionConverters._
+    import scala.jdk.CollectionConverters._
```
Comment (Philippus) [inline _zh-cn/overviews/collections/conversions-between-java-and-scala-collections.md]: This is also for the older collections overview.

File: _overviews/collections/conversions-between-java-and-scala-collections.md (hunk 3 above)
Comment (SethTisue) [reply _overviews/collections/conversions-between-java-and-scala-collections.md]: gah, how'd I miss that, good catch

**Comment by zachhweed** [comment]
Thanks @Philippus I'll get this taken care of.

File: _overviews/collections-2.13/conversions-between-java-and-scala-collections.md (hunk 1 above)
Comment (zachhweed) [reply _overviews/collections-2.13/conversions-between-java-and-scala-collections.md]: Sorry if I'm missing something, but what updated URL should we be using in place of this?

Comment (Philippus) [reply _overviews/collections-2.13/conversions-between-java-and-scala-collections.md]: I guess the link should become `[CollectionConverters](https://scala-lang.org/api/{{ site.scala-version }}/scala/jdk/CollectionConverters$.html)`

**Comment by zachhweed** [comment]
@Philippus thanks for the hand-holding. I was generating the wrong URL at first and got a tad confused, hence my questions :) everything should be good to go now though. Thanks again!

**Pull request merged**